lib/countwords.rb

```diff
@@ -0,0 +1,43 @@
+class CountWords
+
+  attr_accessor :word_count
+
+  def count_words_from_file(file)
+    count_file = File.open(file, "r")
+    count_file.each_line do |line|
+      next if line.empty?
+      @@word_count = Hash.new
+      word = line.split
+      word.each do |wrd|
+        next if wrd !~ /(\w+)/
+        wrd = wrd[/(\w+)/]
+        if @@word_count.has_key?(wrd)
+          @@word_count[wrd] = @@word_count[wrd] + 1
+        else
+          @@word_count[wrd] = 1
+        end
+      end
+      display_word_count_asc
```

Comment:
I think it would be better to not display while you build up the data. How about if the client code (outside this class) asks for the display something like:

```ruby
word_counter = CountWords.new
word_counter.count_words_from_file("speech.txt")
word_counter.display_word_count_asc
```